src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -22,351 +22,358 @@
-          box[j][0] = annotation[j][2]*float(height)
-          box[j][1] = annotation[j][1]*float(width)
-          box[j][2] = (annotation[j][4]+annotation[j][2])*float(height)
-          box[j][3] = (annotation[j][3]+annotation[j][1])*float(width) 
-      
-      confidence = np.expand_dims(annotation[:, 5], axis=1)
-
-      # The bounding box is [n, 4] reshaped and ones added to multiply to tranformation matrix
-      v = np.concatenate([box.reshape(-1, 2), np.ones((box.shape[0]*2, 1), dtype=np.float32)], axis=1)
-      # Transform
-      v = np.dot(v, np.transpose(transformations[i]))
-      # Reverse shape
-      bbox_out = v[:, :2].reshape(-1, 4)
-      
-      # Make points correctly ordered (lower < upper)
-      # Can probably be made much nicer (numpy-ified?)
-      for i in range(len(bbox_out)):
-          if bbox_out[i][0] > bbox_out[i][2]:
-              bbox_out[i][0], bbox_out[i][2] = bbox_out[i][2], bbox_out[i][0]
-          if bbox_out[i][1] > bbox_out[i][3]:
-              bbox_out[i][1], bbox_out[i][3] = bbox_out[i][3], bbox_out[i][1]
-
-          if clip_to_shape is not None:
-              bbox_out[:, 0::2] = np.clip(bbox_out[:, 0::2], 0, clip_to_shape[0])
-              bbox_out[:, 1::2] = np.clip(bbox_out[:, 1::2], 0, clip_to_shape[1])
-
-      bbox = np.zeros(bbox_out.shape)
-      for k in range(len(bbox_out)):
-          bbox[k][0] = bbox_out[k][1]/float(clip_to_shape[0])
-          bbox[k][1] = bbox_out[k][0]/float(clip_to_shape[1])
-          bbox[k][2] = (bbox_out[k][3] - bbox_out[k][1])/float(clip_to_shape[0])
-          bbox[k][3] = (bbox_out[k][2] - bbox_out[k][0])/float(clip_to_shape[1])
-      
-      an = np.hstack((np.hstack((identifier, bbox)), confidence))
-      an = np.ascontiguousarray(an, dtype=np.float32)
-      aug_anns.append(an)
-  return aug_anns
-
-def _assert(cond, ex_type, msg):
-    # A polymorphic assert, works with tensors and boolean expressions.
-    # If `cond` is not a tensor, behave like an ordinary assert statement, except
-    # that a empty list is returned. If `cond` is a tensor, return a list
-    # containing a single TensorFlow assert op.
-
-    if is_tensor(cond):
-       return [control_flow_ops.Assert(cond, [msg])]
-    else:
-        if not cond:
-            raise ex_type(msg)
-        else:
-            return []
-
-def get_augmented_images(images, output_shape):
-
-    # Store transformations and augmented_images for the input batch
-    transformations = []
-    augmented_images = []
-
-    # Augmentation option
-    min_scale = 1/1.5 
-    max_scale = 1.5 
-    max_aspect_ratio=1.5 
-    max_hue=0.05 
-    max_brightness=0.05 
-    max_saturation=1.25 
-    max_contrast=1.25 
-    horizontal_flip=True 
-
-    for i in range(len(images)):
-
-        image = images[i]
-        image = _utils.convert_shared_float_array_to_numpy(image)
+    image_height, image_width, _ = image.shape
+    flipped_image = np.flip(image, 1)
+    
+    for i in range(0, len(annotation)):
```

Comment:
You don't need the `0, `.

More importantly - it's really not clear what this for-loop is doing.

Reply:
One image can have more than one annotation. Will have to do this for all annotations. Can put up comments for this.